experiments/kompanion/cmd/migrator/migrator.go

```diff
@@ -0,0 +1,442 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migrator
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/experiments/kompanion/pkg/utils"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
```

Comment:
We should use `sigs.k8s.io.yaml` when dealing with k8s objects

Reply:
Fixed.